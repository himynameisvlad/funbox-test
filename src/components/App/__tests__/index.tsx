import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';

import { App } from '../index';

jest.mock('../../RouteAside', () => ({
  RouteAside: () => <div />,
}));
jest.mock('../../YandexMap', () => ({
  YandexMap: () => <div />,
}));

describe('App - root component', () => {
  let wrapper;
  const props = {
    moveRoutePoint: jest.fn(),
    loadMapRequest: jest.fn(),
    createMap: jest.fn(),
    setRoutePoints: jest.fn(),
    destroyMap: jest.fn(),
    isMapScriptLoaded: true,
    routePoints: [],
  };
  const AppComponent = <App {...props} />;

  beforeEach(() => {
    wrapper = shallow(AppComponent);
  });

  it('renders correctly', () => {
    const tree = renderer.create(AppComponent).toJSON();

    expect(tree).toMatchSnapshot();
    expect(wrapper.find('RouteAside').length).toBe(1);
    expect(wrapper.find('YandexMap').length).toBe(1);
  });

  it('loads map', () => {
    expect(props.loadMapRequest).toHaveBeenCalled();
  });

  it('renders nothing, if map is not loaded', () => {
    wrapper = shallow(<App {...{ ...props, isMapScriptLoaded: false }} />);

    expect(wrapper.find('main').length).toBe(0);
  });
});
