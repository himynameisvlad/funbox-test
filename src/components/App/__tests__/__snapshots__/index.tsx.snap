// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - root component renders correctly 1`] = `
<main>
  <div
    className="app__aside"
  >
    <div />
  </div>
  <div
    className="app__map"
  >
    <div />
  </div>
</main>
`;
